src/Spark.Mongo/Store/MongoCollectionPageResult.cs

```diff
@@ -44,7 +44,7 @@ public async Task IterateAllPagesAsync(Func<IReadOnlyList<T>, Task> callback)
             for (var offset = 0; offset < TotalRecords; offset += _pageSize)
             {
                 var data = await _collection.Find(_filter)
-                    .Sort(Builders<BsonDocument>.Sort.Ascending(Field.PRIMARYKEY))
+                    .Sort(Builders<BsonDocument>.Sort.Ascending(Field.REFERENCE).Ascending(Field.WHEN))
```

Comment:
This part needs to be reverted

Reply:
Wait a minute, I'm not familiar with the operation of GitHub.